plugin/auth_ldap/src/connection.cc

```diff
@@ -332,8 +332,19 @@ std::string Connection::get_ldap_uri() {
   str_stream << (use_ssl_ ? "ldaps://" : "ldap://") << ldap_host_ << ":"
              << ldap_port_;
   if (!ldap_fallback_host_.empty()) {
-    str_stream << "," << (use_ssl_ ? "ldaps://" : "ldap://")
-               << ldap_fallback_host_ << ":" << ldap_fallback_port_;
+    // We allow 2 formats for fallback:
+    // * only fallback_host is specified (port is 0): in this case, we add it to
+    // the connection string as is
+    // * port is also specified: in this case we add the protocal prefix and
+    // port to it
+    str_stream << ",";
+    if (!ldap_fallback_port_ != 0) {
```

Comment:
this logic does not seems correct for multiple `ldap_fallback_host_` in case we have multiple hosts, only the first one will have ldap[s]:// append to it?

Also, we don't have any validation in terms of if the user adds host with ldap:// already we will append it for a second time.

Reply:
The idea is that we have two configuration options:

* if the user specifies ldap_fallback_port, we tret is as the normal setting, and use multiple variables to build the connection string
* if fallback_port is 0, but fallback_host is specifed, we assume that the user specified one or more complete connection strings in it, and use it as is.

This way the user can specify as many ldap servers as required, and can also mix SSL/TLS and other settings.